src/arch/riscv/pagetable_walker.cc

```diff
@@ -200,18 +200,32 @@ Walker::startWalkWrapper()
 {
     unsigned num_squashed = 0;
     WalkerState *currState = currStates.front();
-    while ((num_squashed < numSquashable) && currState &&
-        currState->translation->squashed()) {
+
+    // check if we get a tlb hit to skip the walk
+    Addr vaddr = Addr(sext<VADDR_BITS>(currState->req->getVaddr()));
+    TlbEntry *e = tlb->lookup(vaddr, currState->satp.asid, currState->mode, true);
+    Fault fault = NoFault;
+    if (e)
+       fault = tlb->checkPermissions(currState->status, currState->pmode, vaddr,
+                                     currState->mode, e->pte);
```

Comment:
Please use `{ }` around multi-line if statements. We've had too many bugs crop up because of this :)

```suggestion
    if (e) {
       fault = tlb->checkPermissions(currState->status, currState->pmode, vaddr,
                                     currState->mode, e->pte);
    }
```